src/hooks.js

```diff
@@ -0,0 +1,41 @@
+import { useState, useEffect } from 'preact/hooks'
+
+const fetchBody = async (url, fetchOpts) => {
+  const state = {}
+  try {
+    state.response = await fetch(url, fetchOpts)
+    state.body = await state.response.text()
+    state.error = state.response.ok
+      ? undefined
+      : Error(state.response.statusText)
```

Comment:
in case of error in the http response, not status `2XX`, fetch doesn't fail.

Because the request was made successfully, even if you get an error back, it's not an issue with the fetch is just that what you get is an http error.

So I build an error from the statusText in that case.

Lated we can use check `state.error` and decide what to do, we still have to body if we have some more info in this error than just the status

Reply:
If the response status is not 204, the response body is read as text and assigned to state.body.

The error state is set based on whether the response was successful:
If the response was successful (state.response.ok is true), state.error is set to undefined.

If the response was not successful (state.response.ok is false), state.error is set to a new Error object with the response's status text.